sam-cli/src/session_engine.rs

```diff
@@ -0,0 +1,104 @@
+use sam_core::engines::SessionSaver;
+use sam_core::entities::choices::Choice;
+use sam_core::entities::identifiers::Identifier;
+use sam_persistence::{SessionError, SessionStorage};
+use std::collections::HashMap;
+use std::path::Path;
+use std::time::Duration;
+use thiserror::Error;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum SessionCommand {
+    Set {
+        var_name: String,
+        choice_value: String,
+    },
+    Clear,
+    List,
+}
+
+pub struct SessionEngine {
+    storage: SessionStorage,
+}
+
+impl SessionEngine {
+    pub fn new(session_dir: impl AsRef<Path>, ttl: Duration) -> Result<Self, ErrorSessionEngine> {
+        let storage = SessionStorage::with_ttl(session_dir, &ttl)?;
+        Ok(SessionEngine { storage })
+    }
+
+    pub fn run(&self, command: SessionCommand) -> Result<i32, ErrorSessionEngine> {
```

Comment:
There is no check for whether the variable that is being set actually exists. 
Therefore a user could set a variable that does not exist, for example if the user makes at typo and this would lead to a bad user experience